src/polygon.typ

```diff
@@ -0,0 +1,59 @@
+/// Returns a list of polygon points from
+/// a list of segments.
+///
+/// Cubic segments get linearized by sampling.
+///
+/// - segment (list): List of segments
+/// - samples (int): Number of samples
+/// -> List of vectors
+#let from-segments(segments, samples: 10) = {
+  import "/src/bezier.typ": cubic-point
+  let poly = ()
+  for ((kind, ..pts)) in segments {
+    if kind == "cubic" {
+      poly += range(0, samples).map(t => {
+        cubic-point(..pts, t / (samples - 1))
+      })
+    } else {
+      poly += pts
+    }
+  }
+  return poly
+}
+
+/// Computes the signed area of a 2D polygon.
+///
+/// The formula used is the following:
+/// $ 1/2 sum_i=0^n-1 x_i*y_i+1 - x_i+1*y_i $
+///
+/// - points (list): List of Vectors of dimension >= 2
+/// -> Signed area
```

Comment:
You need to describe what you mean by a polygon, what form should it take, what expectations does the function make about the polygon.

The equation should be katex compatible, *hideous* I know.

Use `array` not `list`

Just use `float` on the return syntax